# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeParent.cpp
# Commit: ec0e9a72aebb
# Full Hash: ec0e9a72aebb72422b06c1cadee2b3f7dc3538ce
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2020-01-10 05:03:08
# Regressor Bug: 1600472
# File Overlap Count: 2
# Description:
#   Bug 1600472 - Disable allowing sacrificing subpixel anti-aliasing for small screens. r=jrmuizel
#   
#   This patch only allows sacrificing subpixel anti-aliasing when the
#   screen size is larger than WUXGA, and when the force disable pref is not
#   set. In the future, we may also add disable this for high end GPUs.
# ==============================================================================

diff -r 912fed7c5710 -r ec0e9a72aebb gfx/layers/wr/WebRenderBridgeParent.cpp
--- a/gfx/layers/wr/WebRenderBridgeParent.cpp	Thu Jan 09 14:14:08 2020 +0000
+++ b/gfx/layers/wr/WebRenderBridgeParent.cpp	Thu Jan 09 15:23:38 2020 +0000
@@ -358,6 +358,9 @@
     MOZ_ASSERT(api);
     mApis[api->GetRenderRoot()] = api;
   }
+
+  UpdateDebugFlags();
+  UpdateQualitySettings();
 }
 
 WebRenderBridgeParent::WebRenderBridgeParent(const wr::PipelineId& aPipelineId)
@@ -1728,6 +1731,26 @@
   mApis[wr::RenderRoot::Default]->WaitFlushed();
 }
 
+void WebRenderBridgeParent::UpdateQualitySettings() {
+  for (auto& api : mApis) {
+    if (!api) {
+      continue;
+    }
+    wr::TransactionBuilder txn;
+    txn.UpdateQualitySettings(gfxVars::AllowSacrificingSubpixelAA());
+    api->SendTransaction(txn);
+  }
+}
+
+void WebRenderBridgeParent::UpdateDebugFlags() {
+  for (auto& api : mApis) {
+    if (!api) {
+      continue;
+    }
+    api->UpdateDebugFlags(gfxVars::WebRenderDebugFlags());
+  }
+}
+
 #if defined(MOZ_WIDGET_ANDROID)
 void WebRenderBridgeParent::RequestScreenPixels(
     UiCompositorControllerParent* aController) {